subprojects/core/src/main/java/org/gradle/configuration/project/BuildScriptProcessor.java

```diff
@@ -18,24 +18,39 @@
 import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.configuration.ScriptPlugin;
 import org.gradle.configuration.ScriptPluginFactory;
+import org.gradle.internal.event.ListenerManager;
+import org.gradle.internal.resource.local.FileResourceListener;
 import org.gradle.internal.time.Time;
 import org.gradle.internal.time.Timer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.File;
+
 public class BuildScriptProcessor implements ProjectConfigureAction {
     private static final Logger LOGGER = LoggerFactory.getLogger(BuildScriptProcessor.class);
     private final ScriptPluginFactory configurerFactory;
+    private final ListenerManager listenerManager;
 
-    public BuildScriptProcessor(ScriptPluginFactory configurerFactory) {
+    public BuildScriptProcessor(
+        ScriptPluginFactory configurerFactory,
+        ListenerManager listenerManager
+    ) {
         this.configurerFactory = configurerFactory;
+        this.listenerManager = listenerManager;
     }
 
     @Override
     public void execute(final ProjectInternal project) {
         if (LOGGER.isInfoEnabled()) {
             LOGGER.info("Evaluating {} using {}.", project, project.getBuildScriptSource().getDisplayName());
         }
+
+        File buildScriptSourceFile = project.getBuildscript().getSourceFile();
+        if (buildScriptSourceFile != null) {
+            listenerManager.getBroadcaster(FileResourceListener.class).fileObserved(buildScriptSourceFile);
```

Comment:
🎨 Extract method `fileObserved(buildScriptSourceFile)`

Reply:
@bamboo BTW, now I'm thinking - probably if would be better to inject not entire `ListenerManager` here, but just `FileResourceListener`?

Reply:
Makes sense 👍